sdk/core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -219,11 +232,22 @@ public void cancelOperation() throws UnsupportedOperationException {
         // We can not cancel an operation if it was never started
         // It only make sense to call cancel operation if current status IN_PROGRESS.
         if (this.pollResponse != null && this.pollResponse.getStatus() != OperationStatus.IN_PROGRESS) {
-            return;
+            return Mono.empty();
         }
 
         //Time to call cancel
-        this.cancelOperation.accept(this);
+        return this.cancelOperation.apply(this);
+    }
+
+    /**
+     * Returns the final result if the polling operation has been completed.
+     * @return A {@link Mono} containing the final output.
+     */
+    public Mono<R> result() {
+        if (!getStatus().equals(PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED)) {
+            return Mono.error(new IllegalAccessException("The poll operation has not successfully completed."));
```

Comment:
Is IllegalAccess the correct exception?

Reply:
I don't understand enough yet, but almost certainly this is not the right exception type.